src/react/loom-app/header-cell-edit/text-input-submenu.tsx

```diff
@@ -0,0 +1,41 @@
+import React from "react";
+import Submenu from "../../shared/submenu";
+import Flex from "../../shared/flex";
+import Button from "../../shared/button";
+import Input from "../../shared/input";
+import Icon from "../../shared/icon";
+
+interface Props {
+	title: string;
+	value: string;
+	onValueChange: (value: string) => void;
+	onBackClick: () => void;
+}
+
+export default function TextInputSubmenu({
+	title,
+	value,
+	onValueChange,
+	onBackClick,
+}: Props) {
+	const [textValue, setTextValue] = React.useState(value || '')
```

Comment:
What is the purpose of `value || ''`? If the value is falsey, then set the state to an empty string. A falsely value for a string could be "0" or "".

Reply:
I think based on the way that menus are handled, the state should be moved up to the `HeaderMenu` component.

Here is the logic that will take the `localValue` (columnName) and save it in the `loomState`

```javascript
       //header-cell-edit/index.tsx
	React.useEffect(() => {
		if (closeRequest !== null) {
			//If we're on the base menu
			if (submenu === null) {
				if (localValue !== markdown) onNameChange(cellId, localValue);
			}
			onClose();
		}
	}, [
		markdown,
		cellId,
		closeRequest,
		submenu,
		localValue,
		onNameChange,
		onClose,
	]);
```
When a user clicks outside the menu or presses enter, a new close request will be generated and this effect will run.

However, this code could also be implemented into the `text-input-submenu`. That might be a better solution for maintainability in the long term. To do that, you would need to pass the `closeRequest` as a prop and add a `useEffect` similar to this one.

Reply:
I removed the fallback value for textValue.

Also I passed down the `closeRequest` and `onClose` into the `TextInputSubmenu` and gave  the `HeaderMenu` an array with Submenu which will handle the closeRequest themselves. I hope I understood your suggestion with that.